### PR TITLE
Operator chart: added nodeselector, affinity and tolerations

### PR DESCRIPTION
**Pull request by efim-a-efim**
### Change log description

Operator chart: added nodeSelector, affinity and tolerations support

### Purpose of the change

Adds support of scheduling parameters into operator chart.

### What the code does

Adds support of scheduling parameters into operator chart.

### How to verify it
1. Add nodeSelector, affinity or tolerations into operator chart values
2. After installing operator chart, check that they are applied

### CODE REVIEW

**Comment by SrishT** [comment]
@efim-a-efim could u explain the purpose of adding these properties to the operator deployment manifest (also no default value has been configured for these properties i.e. nodeSelector, affinity and tolerations in the values.yaml)

**Comment by efim-a-efim** [comment]
> @efim-a-efim could u explain the purpose of adding these properties to the operator deployment manifest

Yes. In our (and presumably in many other peoples) system we deploy operators to a dedicated namespace and a dedicated set of nodes - it's easier to manage. We use affinities to specify the nodes and taints/tolerations to prevent K8S from launching anything else on that nodes.

> (also no default value has been configured for these properties i.e. nodeSelector, affinity and tolerations in the values.yaml)
Yes, will add.

File: charts/zookeeper-operator/templates/post-install-upgrade-hooks.yaml [hunk 1]
```diff
@@ -105,3 +105,19 @@ spec:
...
+{{ toYaml .Values.nodeSelector | indent 8 }}
+      {{- end }}
+      {{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+      {{- end }}
+      {{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+      {{- end }}
```
Comment (SrishT) [inline charts/zookeeper-operator/templates/post-install-upgrade-hooks.yaml:123]: These parameters do not need to be added to the hook

File: charts/zookeeper-operator/templates/operator.yaml [hunk 2]
```diff
@@ -35,5 +35,23 @@ spec:
               fieldPath: metadata.name
         - name: OPERATOR_NAME
           value: {{ template "zookeeper-operator.fullname" . }}
+        {{- if .Values.resources }}
         resources:
 {{ toYaml .Values.resources | indent 10 }}
+        {{- end }}
+      {{- if default "" (coalesce .Values.imagePullSecrets .Values.global.imagePullSecrets) }}
+      imagePullSecrets:
```
Comment (SrishT) [inline charts/zookeeper-operator/templates/operator.yaml:43]: The imagePullSercret doesnt need to be added here. It has been configured into the service account.

**Review by SrishT — CHANGES_REQUESTED** [review]

File: charts/zookeeper-operator/templates/post-install-upgrade-hooks.yaml [hunk 3]
```diff
@@ -105,3 +105,19 @@ spec:
           configMap:
             name: {{ template "zookeeper-operator.fullname" . }}-post-install-upgrade
             defaultMode: 0555
+      {{- if default "" (coalesce .Values.imagePullSecrets .Values.global.imagePullSecrets) }}
+      imagePullSecrets:
+{{ coalesce .Values.imagePullSecrets .Values.global.imagePullSecrets | toYaml | indent 8 }}
+      {{- end }}
```
Comment (SrishT) [inline charts/zookeeper-operator/templates/post-install-upgrade-hooks.yaml:111]: As in previous case, please remove imagePullSecrets from here

**Review by SrishT — CHANGES_REQUESTED** [review]

**Comment by anishakj** [comment]
@efim-a-efim  Could you please address the review comments.

**Review by SrishT — APPROVED** [review]
LGTM

**Pull request merged**